raiden/network/transport/matrix/transport.py

```diff
@@ -397,6 +397,9 @@ def on_success(greenlet):
         self.greenlets = [self._client.sync_thread]
 
         self._client.set_presence_state(UserPresence.ONLINE.value)
+
+        super().start()  # start greenlet
```

Comment:
This is required to that `_global_send_worker` is invoked before we start sending the other messages

Reply:
This doesn't guarantee that though. Since the listener thread is started above we could AFAIK already be processing (and replying to) new incoming messages. 

One possibility would be to add a check in the retrier to wait for the global queue to become empty. Not sure if that's really a good idea since it could introduce delays in sending non-global messages...